c_src/shmex/nif/shmex/lib.c

```diff
@@ -132,65 +109,24 @@ int shmex_get_from_term(ErlNifEnv * env, ERL_NIF_TERM struct_term, Shmex *payloa
   if (!result) {
     return 0;
   }
-  payload->name = enif_alloc(name_binary.size + 1);
+  payload->name = malloc(name_binary.size + 1);
```

Comment:
~Why have you changed that?~ Ok, I guess you used malloc/free in commons. Anyway, I think we should use `enif_alloc` in NIFs, and that can be easily achieved by defining proper allocator function conditionally in common's header